app/concerns/report_generator/active_storage_download_adapter.rb

```diff
@@ -9,7 +9,7 @@ def expiring_link(expires_in: ReportGenerator::Download::MAX_EXPIRY)
     return self.remote_file_url if self.remote_file_url.present? && !self.report.attached?
 
     if %i[local test].include?(Rails.application.config.active_storage.service)
-      ActiveStorage::Current.host = ReportGenerator.config.local_storage_host
+      ActiveStorage::Current.url_options.host = ReportGenerator.config.local_storage_host
```

Comment:
Think this needs to be:

```suggestion
      ActiveStorage::Current.url_options = { host: ReportGenerator.config.local_storage_host }
```

based on [the source](https://github.com/rails/rails/blob/7-0-stable/activestorage/app/models/active_storage/current.rb#L8)